src/app/core/layers/geolayermixin.js

```diff
@@ -45,7 +45,8 @@ proto.setup = function(config={}, options={}) {
     maxscale: config.maxscale,
     ows_method: config.ows_method,
     exclude_from_legend: (typeof config.exclude_from_legend == 'boolean') ? config.exclude_from_legend : true,
-    categories: false // has more than one categories legend
+    categories: false, // has more than one categories legend,
+    expanded: config.expanded /** Add expanded property useful for legend layer on TOC**/
```

Comment:
```suggestion
    /**
     *  Expand or collapse legend item in catalog layers (TOC)
     *
     * @since v3.8
     */
    expanded: config.expanded
```